src/changelog-check.ts

```
@@ -0,0 +1,200 @@
+
+// Manually define types for changelog-parser since it lacks official TypeScript support
+type Release = {
+  version: string | null;
+  title: string;
+  date: string | null;
+  body: string;
+  parsed: Record<string, string[]>;
+};
+
+type Changelog = {
+  title: string;
+  description: string;
+  versions: Release[];
+};
+
+const changelogParser: (options: {
+  filePath?: string;
+  text?: string;
+  // eslint-disable-next-line @typescript-eslint/no-require-imports, @typescript-eslint/no-var-requires
+}) => Promise<Changelog> = require('changelog-parser');
+
+/**
+ * Asynchronously fetches the CHANGELOG.md file content from a specified GitHub repository and branch.
+ * The function constructs a URL to access the raw content of the file using GitHub's raw content service.
+ * It handles authorization using an optional GitHub token from environment variables.
+ *
+ * @param repo - The full name of the repository (e.g., "owner/repo").
+ * @param branch - The branch from which to fetch the CHANGELOG.md file.
+ * @returns A promise that resolves to the content of the CHANGELOG.md file as a string.
+ * If the fetch operation fails, it logs an error and returns an empty string.
+ */
+async function fetchChangelogFromGitHub(
+  repo: string,
+  branch: string,
+): Promise<string> {
+  // Use raw.githubusercontent.com to directly fetch file content
+  const url = `https://raw.githubusercontent.com/${repo}/${branch}/CHANGELOG.md`;
+  // eslint-disable-next-line n/no-process-env
+  const token = process.env.GITHUB_TOKEN ?? '';
+
+  try {
+    const response = await axios.get(url, {
+      headers: token ? { Authorization: `Bearer ${token}` } : {},
+      responseType: 'text', // Ensure we get raw text
+    });
+
+    return response.data; // Directly return the raw content
+  } catch (error) {
+    console.error(
+      `❌ Error fetching CHANGELOG.md from ${branch} on ${repo}:`,
+      error,
+    );
+    return '';
```

Nit: Why return here rather than throw? It looks like this empty string would get passed along to `parseChangelog, which maybe would fail if given an empty string, not sure. But even there the error is caught and logged, letting the script continue.

We can't check a changelog that doesn't exist

---

Updated to propagate err instead of empty string